File: src/packs/instructions.ts
export interface SitePackInstructions {
  summary: string[];
  raw: string;
}

export function parseInstructions(markdown: string): SitePackInstructions {
  const summary = markdown
    .split('\n')
    .map((line) => line.trim())
    .filter((line) => line.startsWith('- '))
    .map((line) => line.slice(2).trim())
    .filter((line) => line.length > 0);

  return {
    summary,
    raw: markdown
  };
}

File: browser_platform/packs/instructions.py
from __future__ import annotations


def parse_instructions(markdown: str) -> dict:
    summary = [
        line[2:].strip()
        for line in markdown.split('\n')
        if line.strip().startswith('- ')
        and line.strip()[2:].strip()
    ]
    return {'summary': summary, 'raw': markdown}
